pkg/build/admission/defaults/admission.go

```diff
@@ -130,6 +141,18 @@ func (a *buildDefaults) applyBuildDefaults(build *buildapi.Build) {
 			build.Spec.Source.Git.NoProxy = &t
 		}
 	}
+
+	//apply default source secret if one set after all validation
+	//BUG: build fails because secret is not mounted... where to mount it?
```

Comment:
newest cut from master (1.4.0.x). are you setting just a name? 
build.spec.sourceSecret.Name=scmsecret ?